### PR TITLE
DT-2594 Add ability to change role name

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/config/ManageUsersApiExceptionHandler.kt [hunk 1]
```diff
@@ -60,6 +68,21 @@ class HmppsManageUsersApiExceptionHandler {
       )
   }
 
+  @ExceptionHandler(MethodArgumentNotValidException::class)
+  fun handleValidationException(e: MethodArgumentNotValidException): ResponseEntity<ErrorResponse> {
+    val field = if (e.allErrors.size > 0) e.allErrors[0].objectName else "none"
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/config/ManageUsersApiExceptionHandler.kt:73]: Probably not important for this service, but in court-service I did a bit of work to ensure all errors from API where returned to client - I think I added an extra element in the ErrorResponse object.

Ignore for now, but might be important in the future

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/AuthService.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import uk.gov.justice.digital.hmpps.manageusersapi.resource.Role
+import uk.gov.justice.digital.hmpps.manageusersapi.resource.RoleNameAmendment
+
+@Service
+class AuthService(
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/AuthService.kt:13]: Any reason to call this `AuthService` ? Should it be RoleService of something similar?

Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/AuthService.kt:13]: Ah, it goes to Auth - I get it

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/RolesService.kt [hunk 3]
```diff
@@ -1,29 +1,21 @@
...
+import uk.gov.justice.digital.hmpps.manageusersapi.resource.RoleNameAmendment
 
 @Service
 class RolesService(
-  val authWebClient: WebClient,
+  val authService: AuthService,
 ) {
 
   @Throws(RoleNotFoundException::class)
   fun getRoleDetail(roleCode: String): Role {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/RolesService.kt:13]: Outside scope of this PR, but interestingly we could make this reactive and non-blocking :-)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/RolesControllerIntTest.kt [hunk 4]
```diff
@@ -51,6 +58,138 @@ class RolesControllerIntTest : IntegrationTestBase() {
...
+        .put().uri("/roles/OAUTH_ADMIN")
+        .headers(setAuthorisation(roles = listOf("ROLE_ROLES_ADMIN")))
+        .body(BodyInserters.fromValue(mapOf("roleName" to "tim")))
+        .exchange()
+        .expectStatus().isBadRequest
+        .expectBody()
+        .jsonPath("$").value<Map<String, Any>> {
+          assertThat(it["userMessage"] as String).isEqualTo("Validation failure: roleNameAmendment")
+          assertThat(it["developerMessage"] as String).contains("default message [roleName],100,4]")
+        }
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/RolesControllerIntTest.kt:121]: this is what I was mentioning earlier compare to court-register where you end u getting a list of nice messages back

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/AuthService.kt (hunk 2 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/AuthService.kt:13]: yep, the roleservice will be calling out to auth and prison, so pulled out

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/config/ManageUsersApiExceptionHandler.kt (hunk 1 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/config/ManageUsersApiExceptionHandler.kt:73]: Will take a look and see if I can do  something similar

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/RolesService.kt (hunk 3 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/RolesService.kt:13]: Will chat to you about this.

**Pull request merged**